protocol/communities_messenger_test.go

```diff
@@ -4337,3 +4337,101 @@ func (s *MessengerCommunitiesSuite) TestMemberMessagesHasImageLink() {
 	requireMessageWithImage(s.alice, alicePubKey, communityID)
 	requireMessageWithImage(s.bob, alicePubKey, communityID)
 }
+
+func (s *MessengerCommunitiesSuite) sendMention(sender *Messenger, chatID string) *common.Message {
+	ctx := context.Background()
+	messageToSend := common.NewMessage()
+	messageToSend.ChatId = chatID
+	messageToSend.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+	messageToSend.Text = "Hello @" + common.EveryoneMentionTag
+
+	response, err := sender.SendChatMessage(ctx, messageToSend)
+	s.Require().NoError(err)
+	s.Require().Len(response.Messages(), 1)
+	s.Require().True(response.Messages()[0].Mentioned)
+	return response.Messages()[0]
+}
+
+func (s *MessengerCommunitiesSuite) TestAliceDoesNotReceiveMentionWhenSpectating() {
+	// GIVEN: Create an open community
+	community, communityChat := s.createCommunity()
+	s.Require().Len(community.Chats(), 1)
+	s.Require().False(community.Encrypted())
+
+	// Bob joins the community
+	advertiseCommunityTo(&s.Suite, community, s.owner, s.bob)
+	request := &requests.RequestToJoinCommunity{CommunityID: community.ID()}
+	joinCommunity(&s.Suite, community, s.owner, s.bob, request, "")
+
+	// Alice just observes the community
+	advertiseCommunityTo(&s.Suite, community, s.owner, s.alice)
+	_, err := s.alice.SpectateCommunity(community.ID())
+	s.Require().NoError(err)
+
+	aliceCommunity, err := s.alice.GetCommunityByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().Len(aliceCommunity.Chats(), 1)
+
+	// Owner creates a new channel
+	newChannel := &protobuf.CommunityChat{
```

Comment:
Alice doesn't get a message from B when I remove this part. Any advice on why this can happen is welcome

(described in the comment https://github.com/status-im/status-go/pull/5180#discussion_r1608439306)